language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -507,6 +518,32 @@ renderDecoder = \case
     DecoderRef t -> snd (genType t) <> ".decoder"
     DecoderLazy d -> "damlTypes.lazyMemo(function () { return " <> renderDecoder d <> "; })"
 
+data Encode
+    = EncodeRef TypeRef
+    | EncodeVariant [(T.Text, TypeRef)]
+    | EncodeAsIs
+    | EncodeRecord [(T.Text, TypeRef)]
+    | EncodeThrow
+
+renderEncode :: Encode -> T.Text
+renderEncode = \case
+    EncodeRef ref -> let (_, companion) = genType ref in
+        "function (__typed__) { return " <> companion <> ".encode(__typed__); }"
+    EncodeVariant alts -> T.unlines $ concat
+        [ [ "function (__typed__) {" ]
+        , [ "  if (__typed__.tag === '" <> name <> "') { return {tag: __typed__.tag, value: " <> companion <> ".encode(__typed__.value) }; }"
```

Comment:
A switch would be a bit clearer imho and perhaps more efficient (hopefully not but I don’t know how good JS JITs are here).

Reply:
Will do.